packages/react/src/reactrouterv6.tsx

```diff
@@ -94,13 +108,31 @@ function getNormalizedName(
 
   const branches = matchRoutes(routes, location);
 
+  let pathBuilder = '';
   if (branches) {
     // eslint-disable-next-line @typescript-eslint/prefer-for-of
     for (let x = 0; x < branches.length; x++) {
-      if (branches[x].route && branches[x].route.path && branches[x].pathname === location.pathname) {
-        const path = branches[x].route.path;
+      const branch = branches[x];
+      const route = branch.route;
+      if (route) {
+        // Early return if index route
+        if (route.index) {
+          return [branch.pathname, 'route'];
+        }
+
+        const path = route.path;
         if (path) {
-          return [path, 'route'];
+          const newPath = path[0] === '/' ? path : `/${path}`;
+          pathBuilder += newPath;
+          if (branch.pathname === location.pathname) {
+            // If the route defined on the element is something like
+            // <Route path="/stores/:storeId/products/:productId" element={<div>Product</div>} />
+            // We should check against the branch.pathname for the number of / seperators
+            if (pathBuilder.split('/').length !== branch.pathname.split('/').length) {
```

Comment:
Is this a safe check? Just asking because I used URL segment counting in my Express router instrumentation and I head to handle a few edge cases. For example, could there be problems with routes starting or ending with a `/` while others don't? (which would influence the length because of empty array items after the .split operation). 

Just a suggestion but now that I'm thinking about it, we could extract this function

https://github.com/getsentry/sentry-javascript/blob/6ea53edb34dc88f708eabc0c083d2fce04f5cbcf/packages/tracing/src/integrations/node/express.ts#L387-L394

to our utils package and use it here. But totally optional ofc.

Reply:
Great point! I'm going to extract out `getNumberOfUrlSegments` and use it instead.